src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -56,13 +62,22 @@ class ContentExplorer extends Component {
          * @param {Array} newFoldersPath
          */
         onEnterFolder: PropTypes.func.isRequired,
+        /** Called when the folders path is updated */
+        onFoldersPath: PropTypes.func,
```

Comment:
is this duplicating the use of `onEnterFolder`? do we need both props?

Reply:
It covers a little more than entering a folder. The parent component needs to know the most up to date folderspath so that it knows where it is in relation to the folder chosen to deep scan with (aka the folder chosen while the toggle is on). onEnterFolder just handles when you click on a folder or a breadcrumb, but not when you go back to the all files modal page

I might be able to use a combination of onEnterFolder and something else to mimic on Folderspath

Reply:
that sounds like a bug if `onEnterFolder` is not updating when the user goes back to the root folder. based on the description, it should update whenever the current folder changes

Reply:
Also, onEnterFolder takes in just a single folder. I need the whole foldersPath and I dont know if I can alter the function to take in the folderspath as well since other components could be using it

Reply:
I think it should be available as the second param:

https://github.com/box/box-ui-elements/blob/3689909192a6626cab6356007bdd29c83b83ebe7/src/features/content-explorer/content-explorer/ContentExplorer.js#L287

https://github.com/box/box-ui-elements/blob/3689909192a6626cab6356007bdd29c83b83ebe7/src/features/content-explorer/content-explorer-modal-container/ContentExplorerModalContainer.js#L172

although it's possible there's a bug somewhere

Reply:
Yeah so we can actually pass up the newFoldersPath which is exactly what I want. The only thing is that the folder nor the folders path is being passed up when the user clicks on the Search Results breadcrumb. In this function https://github.com/box/box-ui-elements/blob/master/src/features/content-explorer/content-explorer/ContentExplorerHeaderActions.js#L63-L82
return is called instead of onEnterFolder if the user clicks on the Search Results

Reply:
gotcha, I think onFoldersPath is good then